### ISSUE TITLE
Feature Proposal: If a fact/events contains a TYPE, show TYPE label instead of label of fact/event

### ISSUE THREAD

**Issue by Jefferson49** [fisharebest/webtrees#4982]
In a discussion in the webtrees forum, [Confirmation - German Translation](https://www.webtrees.net/index.php/forum/8-translation/38919-confirmation-german-translation), the following feature was proposed:

**If a fact/events contains a TYPE, the fact/event view should show the TYPE label instead of the fact/event label.**

This feature is already implemented for some GEDCOM structures:
[github.com/fisharebest/webtrees/blob/9fd...p/Fact.php#L452-L460](https://github.com/fisharebest/webtrees/blob/9fdb55cab6066dd20ca53c01a3aa77db35c85483/app/Fact.php#L452-L460)
[github.com/fisharebest/webtrees/blob/9fd...s/fact.phtml#L67-L71](https://github.com/fisharebest/webtrees/blob/9fdb55cab6066dd20ca53c01a3aa77db35c85483/resources/views/fact.phtml#L67-L71)

E.g., the following GEDCOM code is already handled accordingly.

1 MARR
2 TYPE Not married

1 EVEN
2 TYPE Military Service

One approach could be that the existing implementation could be generalized for all tag combinations **INDI:*:TYPE**

To give some room for individual likings of the users, it might be helpful to also add some preferences to the control panel. Maybe, something like the attached image, where FACT/EVEN could be used as default values:

![Preferences TYPE view](https://github.com/fisharebest/webtrees/assets/81484983/894eeae2-46fc-42bb-8fe2-f10d4d52591d)

**Comment by fisharebest** [comment]
Even if we make this completely-configurable, we will still need default values for which tags have this feature and which do not.

**Comment by Jefferson49** [comment]
> Even if we make this completely-configurable, we will still need default values for which tags have this feature and which do not.

Yes, I agree. My proposal would be that INDI:EVEN:TYPE and INDI:FACT:TYPE should be used as a default. This would match the currently available implementation, which already shows types in these cases.

For me personally, configuration is just and add on option. We could also consider a 2 step approach. In a first step, we could start without any configuration and show the type label for all facts/events. If there is some critical feedback from the users, we could add some configuration in a second step.

**Comment by Norwegian-Sardines** [comment]
Not all “facts” end their descriptive path with the TYPE tag!

Before this is put into action, please read my suggestion for GEDCOM v7.0.x

https://github.com/FamilySearch/GEDCOM/pull/301#issuecomment-1513403034

**Comment by fisharebest** [comment]
For most (all?) of our `TYPE` fields, the translation is lowercase, so that it can be used in a `Label: value` string such as `Type: foo`.

e.g.

```
1 NAME xxx
2 TYPE married name
```

becomes in spanish

> Nombre: xxx
> Tipo: nombre de casado

Thus we can't use it directly as the label.  It would need to be "Nombre de casado" instead of "nombre de casado".

**Comment by Jefferson49** [comment]
> the translation is lowercase, so that it can be used in a Label: value string such as Type: foo

Yes, this is an issue for the name types from the **NameType** class, which are currently in lower case. 

I am not an English or Spanish native speaker, but would it be wrong if they start with an upper case anyway? For me,  `Type: Foo` looks acceptable. And birth name starting with an upper case in the name view might be o.k:

![birth name](https://github.com/fisharebest/webtrees/assets/81484983/7663f848-a30a-47a9-b5b8-9f9e0c7fd746)

However, one big disadvantage would be that all the translation would need to be changed as well. Probably difficult to organize.

I also checked the **MarriageType** class. The 3 included types start with an upper case.

I also tried to identify other Type classes in the \Elements folder (see screenshot below). It looks like they might not be that relevant. However, it would also need a decision how to proceed with these types. My feeling would be to limit the label issue to INDI:*:TYPE and FAM:MARR:TYPE.

![Elements Types](https://github.com/fisharebest/webtrees/assets/81484983/56182fb2-1f6a-4497-9f7b-2503436f4d63)

**Comment by Norwegian-Sardines** [comment]
In GEDCOM v7.0.x,  all TYPE enumerations are all uppercase!  At some point in the future this will need attention.   Is it possible for a future release to have translations for TYPE to have multiple options?  For example:  INDI.NAME.TYPE = “MARRIED”, translates to (sentence case) “Married”, (lower case) “married”.  

For multi-word enumerations (AKA) we would maybe need to eliminate(sentence case) and call it (title case) translate to “Also Known As”.

Just some thoughts.
